service/history/shardController_test.go

```diff
@@ -262,12 +264,12 @@ func (s *shardControllerSuite) TestAcquireShardLookupFailure() {
 	numShards := 2
 	s.config.NumberOfShards = numShards
 	for shardID := 1; shardID <= numShards; shardID++ {
-		s.mockServiceResolver.EXPECT().Lookup(string(shardID)).Return(nil, errors.New("ring failure")).Times(1)
+		s.mockServiceResolver.EXPECT().Lookup(convert.IntToString(shardID)).Return(nil, errors.New("ring failure")).Times(1)
```

Comment:
For all those mocks you need to do something like this:
```suggestion
		shardIDstr := convert.IntToString(shardID)
		s.mockServiceResolver.EXPECT().Lookup(shardIDstr).Return(nil, errors.New("ring failure")).Times(1)
```